src/main/java/com/carrotcreative/cream/util/HashingUtil.java

```diff
@@ -0,0 +1,60 @@
+package com.carrotcreative.cream.util;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import android.support.annotation.NonNull;
+
+public class HashingUtil
+{
+    private static final int CACHE_SIZE = 50;
+    private static LruCache<String, String> mCache = new LruCache<>(CACHE_SIZE);
+
+    private HashingUtil() {}
+
+    /**
+     *
+     * @param secret The string that needs to be hashed.
+     * @return The hashed value.
+     */
+    public static String genHash(@NonNull String secret)
```

Comment:
Hmm..  Could we possibly rename this to `getHash`?

It seems like here we're getting the hash for the value, and the other function is actually generating it.